_transcripts/039-rust-1.54-1.55.md

```diff
@@ -0,0 +1,718 @@
+stepping stone into getting to the actual `Try` trait.
+
+There was one other stabilized API that is not super interesting, but I found it
+a little curious, which is `Drain::as_str`. So this is— so you may realize that
+if you have a vector, you can call `drain` on it in order to remove all the
+elements indicated by the range that you passed to `drain`, but keep the vector
+otherwise intact. So for example if you have a vector of length five, you could
+say, I want to drain elements 2 through 4, and it gives you an iterator of the
+owned elements from 2 to 4. And what it leaves behind in the vector is elements
+one and five. Those are both left in the vector and they're sort of shifted
+around so that they remain together. Well, strings, as in `String`, also have a
+`drain` method that do sort of the same thing. You give it a range of characters
+inside the `String`, and you say, drain those characters out of the `String`,
+and then leave the remaining characters intact. And when you call `drain` you
+get back a new type called `Drain`, and that's the thing that implements
+`Iterator`. And that `Drain` type now has gained a method called `as_str`. And
+the idea here is that if you drain characters out of a string, the `Drain` can
+give you a `str` reference to the characters you have yet to drain. And that's
+what was added.
+
+And it sort of makes sense, right? Like, you're removing a— you're basically
+removing a `str` from a `String`. And so if you've removed some of the
+characters, this lets you get at, what is the remainder of things that I haven't
+removed yet, as a `str`. It's just like a— it gives you an interesting insight
+into what what `Drain` does, and sort of, the uses that you might have for it.
+
+So I think that's all we had for the stabilized APIs. So I think all that's left
+is my usual deep dive into the into the changelog. And there's not too much of
+interest there, I think, for 1.55. There is one change, which is the build
+scripts are now told about `RUSTFLAGS` and the `rustc` wrapper and stuff. So
+this is, if your cargo configuration includes sending additional flags to
+`rustc`, or has a wrapper around `rustc`, previously build scripts weren't
+informed of that, and there are some crates that use build scripts to do things
+like, determine whether they can use a nightly feature or not. And so they would
+break if you had Rust flags that they didn't take into account. That's passed
+in, which is kind of nice.
+
+Another one is that we now have `cargo clippy --fix` stabilized. So we've had
+`cargo fix` for a while, for things like edition changes. But now— or even just
+like if the compiler can automatically fix a given warning or error, I think you
+can pass, like `cargo --fix` or just `cargo fix`, and now we have the same for
+Clippy. So if Clippy detects a lint that it thinks it can— it has like, an
+automated fix for, you can just call `cargo clippy --fix` and it will fix those
```

Comment:
So `cargo --fix` is wrong, and `cargo fix` is correct? I'm not sure how to format this; I could do something like:

> like ~~"cargo dash-dash-fix"~~ or just `cargo fix`

Or do the heavy-handed thing and just delete the wrong one entirely?

(38:55)

Reply:
I like dash-through for this